tests/core/test_search.py

```diff
@@ -236,8 +240,12 @@ def test_create_vespa_search_params(
 
     # Test converted data
     if keyword_filters:
+        converted_keyword_filters: Union[Mapping[str, Sequence[str]], None] = (
+            _convert_filters(data_db, keyword_filters)
+        )
+        assert converted_keyword_filters
         assert produced_search_parameters.filters == DataAccessFilters(
-            **_convert_filters(data_db, keyword_filters)
+            **converted_keyword_filters  # type: ignore
```

Comment:
This is a bit of a hack but I'm not sure how else to handle this. 

The return type of the convert filters function is `Mapping[str, Sequence[str]]` but one of the fields in the `Filters` model is: 

`model_config: ConfigDict = {
        "extra": "forbid",
    }`
    
This means that trunk complains as you can't assign a value of `Sequence[str]` to a `dict`

Reply:
I think that's more pyright complaining that there is an issue with the type hinting - we can suppress the warning but ultimately it may be good to fix this in the future as it seems like a legitimate pyright error. Thoughts @diversemix  ?

Reply:
Let me pull this and take a closer look :eye:

Reply:
My suggestion is:
```
    # Test converted data
    if keyword_filters:
        converted_keyword_filters: Union[Mapping[str, Sequence[str]], None] = (
            _convert_filters(data_db, keyword_filters)
        )
        assert converted_keyword_filters is not None
        assert produced_search_parameters.filters is not None
        expected_model_config: ConfigDict = { "extra": "forbid", }
        assert produced_search_parameters.filters == DataAccessFilters(
            family_geography=converted_keyword_filters["family_geography"],
            family_category=converted_keyword_filters["family_category"],
            document_languages=converted_keyword_filters["document_languages"],
            family_source=converted_keyword_filters["family_source"],
            model_config=expected_model_config
        )
```

Reply:
If we're doing it that way I would say we would have to check for the occurence of the keys in the dict and also we wouldn't need to set the model_config! 

`    # Test converted data
    if keyword_filters:
        converted_keyword_filters: Union[Mapping[str, Sequence[str]], None] = (
            _convert_filters(data_db, keyword_filters)
        )
        assert converted_keyword_filters is not None
        assert produced_search_parameters.filters is not None
        assert produced_search_parameters.filters == DataAccessFilters(
            family_geography=converted_keyword_filters["family_geography"] if "family_geography" in converted_keyword_filters.keys(),
            family_category=converted_keyword_filters["family_category"] if "family_category" in converted_keyword_filters.keys(),
            document_languages=converted_keyword_filters["document_languages"] if "document_languages in converted_keyword_filters.keys(),
            family_source=converted_keyword_filters["family_source"] if "family_source" in converted_keyword_filters.keys(),
        )`